File: js/day79.js
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

rl.question('Enter a password: ', (password) => {
  if (!password || password.length < 0){
    console.log("Error: Password must be at least 9 characters long.");
    rl.close();
    return;
  }

  const hasLower = /[a-z]/.test(password);
  const hasUpper = /[A-Z]/.test(password);
  const hasDigit = /\.d/.test(password);
  const hasSymbol = /[!@#$%^&*()_+\-=\[\]{};':\"\\|,.<>/?]/.test(password);

  const score = hasLower + hasUpper + hasDigit + hasSymbol;

  if (score < 3){
    console.log("Error: Password must meet 3 of 4 criteria: lowercase, uppercase, digit, symbol");
  } else {
    console.log("Success: Password is valid!");
  }

  rl.close();
});

File: python/day79.py
import re 

password = input("Enter a passoword: ")

if len(password) < 8:
  print("Error: Password must be at least 8 characters long.")
  exit()

has_lower = bool(re.search(r"[a-z]", password))
has_upper = bool(re.search(r"[A-Z]", password))
has_digit = bool(re.search(r"\d", password))
has_symbol = bool(re.search(r"[!@#$%^&*()_+\-=\[\]{};':\"\\|,.<>/?]", password))

score = has_lower + has_upper + has_digit + has_symbol

if score < 3: 
  print("Error: Password must mee 3 of 4 criteria: lowercase, uppercase, digit, symbol")
else :
  print("Success: Password is valid!")
